Ifpack2: Build errors in block relaxation unit test & others, when Epetra is ON but AztecOO is OFF

@trilinos/ifpack2 

Ifpack2's BlockRelaxationPerformance unit test has build errors.  Here are some of them:
```
.../Trilinos/packages/ifpack2/test/unit_tests/Ifpack2_UnitTestBlockRelaxationPerf.cpp:148:5: error: unknown type name 'Epetra_MpiComm'; did you mean 'Epetra_Comm'?
    Epetra_MpiComm ecomm(MPI_COMM_WORLD);
    ^~~~~~~~~~~~~~
    Epetra_Comm
.../Trilinos/packages/ifpack2/test/unit_tests/../../../ml/src/Utils/ml_epetra_utils.h:18:7: note: 'Epetra_Comm' declared here
class Epetra_Comm;
      ^
.../Trilinos/packages/ifpack2/test/unit_tests/Ifpack2_UnitTestBlockRelaxationPerf.cpp:148:20: error: variable type 'Epetra_Comm' is an abstract class
    Epetra_MpiComm ecomm(MPI_COMM_WORLD);
                   ^
```
There are also some build errors in `ifpack2/example/RelaxationWithEquilibration.cpp`.  It looks like these come up if Epetra is enabled but AztecOO is disabled.  That's unusual, but possible.

--- a/packages/ifpack2/example/RelaxationWithEquilibration.cpp
+++ b/packages/ifpack2/example/RelaxationWithEquilibration.cpp
@@ -18,8 +18,11 @@
 #include "KokkosBlas1_nrm2.hpp"
 #include "KokkosBlas3_gemm.hpp"
 
-#if defined(HAVE_IFPACK2_AZTECOO) && defined(HAVE_IFPACK2_EPETRA)
+#if defined(HAVE_IFPACK2_AZTECOO)
 #  include "AztecOO.h"
+#endif // defined(HAVE_IFPACK2_AZTECOO)
+
+#if defined(HAVE_IFPACK2_EPETRA)
 #  include "Epetra_Comm.h"
 #  if defined(HAVE_MPI)
 #    include "mpi.h"
@@ -33,7 +36,7 @@
 #  if defined(HAVE_TPETRACORE_MPI)
 #    include "Tpetra_Details_extractMpiCommFromTeuchos.hpp"
 #  endif // HAVE_TPETRACORE_MPI
-#endif // defined(HAVE_IFPACK2_AZTECOO) && defined(HAVE_IFPACK2_EPETRA)
+#endif // defined(HAVE_IFPACK2_EPETRA)
 
 #include <algorithm> // std::transform
 #include <cctype> // std::toupper

--- a/packages/ifpack2/test/unit_tests/Ifpack2_UnitTestBlockRelaxationPerf.cpp
+++ b/packages/ifpack2/test/unit_tests/Ifpack2_UnitTestBlockRelaxationPerf.cpp
@@ -11,7 +11,11 @@
 #include "Epetra_Vector.h"
 #include "Epetra_CrsMatrix.h"
 #include "Epetra_LinearProblem.h"
-#include "Epetra_SerialComm.h"
+#if defined(EPETRA_MPI)
+#  include "Epetra_MpiComm.h"
+#else
+#  include "Epetra_SerialComm.h"
+#endif // defined(EPETRA_MPI)
 
 #include "ml_include.h"
 #include "ml_MultiLevelPreconditioner.h"
@@ -144,7 +148,7 @@ TEUCHOS_UNIT_TEST(Ifpack2BlockRelaxation, Performance)
     auto tmat = create_testmat_t(tmap);
 
     myOut << "Create Epetra test matrix" << std::endl;
-#ifdef EPETRA_MPI
+#if defined(EPETRA_MPI)
     Epetra_MpiComm ecomm(MPI_COMM_WORLD);
 #else
     Epetra_SerialComm ecomm;

